app/src/main/java/com/example/farmeraid/data/CharityRepository.kt

```diff
@@ -57,17 +60,19 @@ class CharityRepository (
         if (!docRead.exists()) {
             return ResponseModel.FAResponseWithData.Error("Charity does not exist")
         }
+        
+        val items = docRead.data?.get("produce") as MutableMap<String, Int>?
+        val name  = docRead.data?.get("charityName") as String?
+        val location = docRead.data?.get("location") as String?
+        val coordinates = docRead.data?.get("coordinates") as GeoPoint?
 
-        val items = docRead.data?.get("produce")
-        val name  = docRead.data?.get("charityName")
-        val location = docRead.data?.get("location")
-
-        return if (items != null && name !=null && location!=null) {
-            val itemsMap = items as MutableMap<String, Int>
-            val fridgeName = name as String
-            val charityLocation = location as String
+        return if (items != null && name !=null && location!=null && coordinates != null) {
+            val itemsMap = items
+            val fridgeName = name
+            val charityLocation = location
             ResponseModel.FAResponseWithData.Success(
                 FridgeModel.Fridge(
+                    id = id,
```

Comment:
I realized my changes hasnt come here yet, but I will rebase my pr to support the geo stuff